.github/workflows/snyk.yml

```diff
@@ -0,0 +1,67 @@
+name: "Snyk"
+
+on:
+  # Please uncomment the push event for testing purposes
+  #push:
+  #  branches: [main]
+  # Run the job at the end of every day
+  schedule:
+    - cron: "0 0 * * *"
+
+env:
+  # Please check how to create an access token here https://docs.snyk.io/tutorials/amazon-web-services/aws-code-suite/snyk-security/create-account-and-obtain-a-token
+  SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+  DEFAULT_JDK_VERSION: 11
+
+jobs:
+  quarkus:
+    name: Quarkus
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: "${{ env.DEFAULT_JDK_VERSION }}"
+      
+      - name: Build Keycloak
+        run: mvn clean install -nsu -B -e -Pdistribution -Dmaven.test.skip -DskipQuarkus -DskipTestsuite -DskipExamples -DskipTests
+
+      - uses: snyk/actions/setup@master
+      - name: Check for vulnerabilities
+        run: snyk test --policy-path=.snyk --all-projects --prune-repeated-subdependencies --exclude=tests --sarif-file-output=quarkus-report.sarif quarkus
+        continue-on-error: true
+        env:
+          SNYK_TOKEN: "${{ env.SNYK_TOKEN }}"
+
+      - name: Upload scanner results to GitHub
+        uses: github/codeql-action/upload-sarif@v1
+        with:
+          sarif_file: quarkus-report.sarif
+
+  operator:
+    name: Operator
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: "${{ env.DEFAULT_JDK_VERSION }}"
+      
+      - name: Build Keycloak
+        run: mvn clean install -nsu -B -e -Pdistribution -Dmaven.test.skip -DskipQuarkus -DskipTestsuite -DskipExamples -DskipTests
```

Comment:
Do we really need to build KC twice?

Reply:
I tried to reuse the build steps and failed. After looking at https://github.com/keycloak/keycloak/blob/main/.github/workflows/ci.yml I noticed that we don't do anything different there. But if you have a solution, I will be more than happy to incorporate the changes here.

Reply:
We could use the same trick as with tests – build KC in a separate job and archive the artifacts:
https://github.com/keycloak/keycloak/blob/19e19c9ddad64d23a29bb2eb3ec740187edc13c6/.github/workflows/ci.yml#L13-L53

Reply:
@vmuzikar introduced Maven cache as @jonkoops suggested. I don't mind doing any optimizations, but considering that this job will run only once a day, we can improve as we go.
If we identify any issues, we can perform more optimizations.

Reply:
👍